core/main/src/service/apps/delegated_launcher_handler.rs

```diff
@@ -95,6 +95,8 @@ pub struct App {
 #[derive(Debug, Clone, Default)]
 pub struct AppManagerState {
     apps: Arc<RwLock<HashMap<String, App>>>,
+    // Very useful for internal launcher where the intent might get untagged
+    intents: Arc<RwLock<HashMap<String, NavigationIntent>>>,
```

Comment:
Can't you add an `intent` field to `App` so tat you can just have the one existing `apps` object?

Reply:
App gets created using the Session call intents is stored temporarily till that call arrives and replace the intent